concent_api/signing_service_integration_test_recovery.py

```diff
@@ -0,0 +1,246 @@
+from signing_service_testing_common import Components
+from signing_service_testing_common import create_golem_message_frame_with_correct_transaction_signing_request
+from signing_service_testing_common import run_tests
+from signing_service_testing_common import send_message_to_middleman_and_receive_response
+from signing_service_testing_common import send_message_to_middleman_without_response
+
+
+REQUIRED_COMPONENTS = []  # type: ignore
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+
+def run_middleman() -> subprocess.Popen:
+    return subprocess.Popen(
+        [
+            'python',
+            'manage.py',
+            'middleman'
+        ]
+    )
+
+
+def run_signing_service() -> subprocess.Popen:
+    return subprocess.Popen(
+        [
+            'python',
+            '-m',
+            'signing_service.signing_service',
+            '127.0.0.1',
+            '"85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=="',
+            '--signing-service-private-key',
+            '"GfkWzCRGL/Me2wk41mS0NRSch2XMpgIXYLRaasmcozw="',
+            '--ethereum-private-key',
+            '"M2ExMDc2YmY0NWFiODc3MTJhZDY0Y2NiM2IxMDIxNzczN2Y3ZmFhY2JmMjg3MmU4OGZkZDlhNTM3ZDhmZTI2Ng=="'
+        ],
+        cwd='../signing_service/'
+    )
+
+
+@count_fails
+def test_case_0_prove_of_concept(config: ConfigParser, **kwargs: Any) -> None:
+    """
+    1. Spawn MiddleMan and SigningService processes.
+    2. Client sends GolemMessageFrame with correct TransactionSigningRequest to MiddleMan
+       and receives response to prove that connection works.
+    """
+
+    try:
+        middleman_process = run_middleman()
+        signing_service_process = run_signing_service()
+
+        # Waiting for MiddleMan and SigningService to start.
+        sleep(10)
+
+        # Create GolemMessageFrame with correct TransactionSigningRequest.
+        golem_message_frame = create_golem_message_frame_with_correct_transaction_signing_request(
+            request_id=get_current_utc_timestamp(),
+        )
+
+        # Send message through wrapper and receive deserialized response.
+        response = send_message_to_middleman_and_receive_response(
+            message=golem_message_frame,
+            config=config,
+            concent_private_key=CONCENT_PRIVATE_KEY,
+            concent_public_key=CONCENT_PUBLIC_KEY,
+        )
+
+        # Check response.
+        assert_condition(type(response), GolemMessageFrame, f'Deserialized response type is {type(response)} instead of GolemMessageFrame.')
+        assert_condition(type(response.payload), SignedTransaction, f'Deserialized response payload type is {type(response.payload)} instead of SignedTransaction.')
+        assert_condition(response.request_id, golem_message_frame.request_id, f'Deserialized response request_id is {response.request_id} instead of {golem_message_frame.request_id}.')
+
+    except Exception as exception:
```

Comment:
I think @cameel would not be happy about putting such big portion of code in `try/except` block.

Reply:
Ok guys, let me explain the main problem with decorator - it would fit only for first test (which is not really a test but a prove of concept that MiddleMan and SigningService will communicate when spawn as processes inside test). In other tests, I am first creating instances 
```
middleman_process = run_middleman()
signing_service_process = run_signing_service()
```

but then in the middle of the test, I am killing and recreating one of them
```
middleman_process.kill()
...
middleman_process = run_middleman()
```

So sometimes there would be the same instance in `try` and `finally` blocks (if something goes wrong before recreating) but in most cases, including successful test, it would be a different one. Then it would try to kill already killed process, leaving another one not killed.
 
About re-raising exceptions you are right, I have removed those :+1: done